py_scripts/simple_pub.py

```diff
@@ -115,19 +123,35 @@
     node_id = publisher_id
     pub_key_id = subscriber_id
 
+event = threading.Event()
+
 def on_ack(pub, payload):
     print "Ack for pub UUID %s(%d)" % (dps.publication_get_uuid(pub), dps.publication_get_sequence_num(pub))
     print "    %s" % (payload)
 
+def on_link(node, addr, status):
+    print "Publisher is linked to %s" % (addr)
+    event.set()
+
 def on_destroy(node):
     print "Destroyed"
     dps.destroy_key_store(key_store)
 
 node = dps.create_node("/", key_store, node_id)
-dps.start_node(node, dps.MCAST_PUB_ENABLE_SEND, 0)
+dps.start_node(node, dps.MCAST_PUB_ENABLE_SEND, args.listen)
 print "Publisher is listening on port %d" % (dps.get_port_number(node))
-pub = dps.create_publication(node)
 
+if args.host != None or args.port != 0:
```

Comment:
Looks like this allows port 0 to be used which is probably not what you intended

Reply:
Right, I want 0 to be valid only on the listening side.  I'll fix that up.